pyprep/find_noisy_channels.py

```diff
@@ -518,6 +518,8 @@ def find_bad_by_ransac(
                 if len(chunk_sizes):
                     chunk_size = chunk_sizes.pop()
                 else:
+                    # Can't test this branch since the '1 channel at a time' case
+                    # was already verified before
```

Comment:
I would remove this comment. If you wanted to test that branch, it'd be possible - you'd just have to construct an appropriate test. (you don't have to, but you could, so the comment is unnecessary IMHO)

```suggestion
```

Reply:
I was trying to make a test but couldn't figure one easy enough. The problem is that if there is a MemoryError at 1 channel at a time it will be caught at line:

https://github.com/sappelhoff/pyprep/blob/87d512581e90cb9a17a88f4a413bd339f015e13d/pyprep/find_noisy_channels.py#L448

So to test the that part I would need to bypass that line somehow; I guess in real life that would be because the amount of ram available changed, so it is possible to get to that line. I could try to implement a code that bypasses it artificially with a flag as an input argument to the find_bad_by_ransac function but that sort of seems a bit off...

Do you have any idea on how to circumvent this?

Reply:
> I guess in real life that would be because the amount of ram available changed, so it is possible to get to that line.

I see :thinking: so it's possible to trigger ... but not easily doable in a CI setting.

> with a flag as an input argument to the find_bad_by_ransac function but that sort of seems a bit off...

that seems off, yes.

--> for these cases we can be pragmatic and use `pragma: no cover`, so that part of the code does not count into our coverage report. 

see this example:
```python
def dumb_function(a, b):
    a = 1
    if a == 1:
        print(b)
    else:  # pragma: no cover
        print("this can never be triggered")
        print("but 'pragma: no cover' takes care that it does not negatively affect our coverage metric ...")
        print("... it is valid for all lines happening under this 'else' nesting")
```